### ISSUE TITLE
[Question] Even distribution between spawnpoints

### ISSUE THREAD

**Issue by Microbicus** [dwesterwick/SPTQuestingBots#38]
Intended or not, pmc spawns are always distributed in a way that they are on the opposite side of the map from player. On Customs it is especially noticable e.g. spawning near green means that all pmc spawn near Tarcone (and die there) and vice versa with only a few roaming the map.

Is there an option to add setting to use spawnpoints "randomly"?

Mods used: SAIN (with deathwish preset)+ LootingBots + FIKA + all dependencies.

**Comment by dwesterwick** [comment]
Depending on what you're experiencing, this might be on purpose. When this mod selects a location to spawn a bot group, it selects the EFT spawn point furthest from any other human or "human" AI (PMC's and PScavs). However, it performs this check for every bot group it spawns. Practically speaking, this means that if you spawn near Big Red on Customs, the first bot group that spawns will be on the complete opposite side of the map. The second will be between you and the first bot group (but on the outskirts of the map), and the cycle continues until there are no other bot groups to spawn. In my experience, this means that if you spawn near Big Red, there will be at most one other PMC spawn on your side of the map. All other PMC spawns will be distributed fairly equally on the other side of the river. 

If you're seeing multiple bot groups close to each other at the beginning of a raid, something seems wrong. To troubleshoot, I will need your LogOutput.log file after you next reproduce this problem. 

Based on how the spawning system is designed, there is no way to specify spawn points (either explicitly or via a distribution).

**Comment by Microbicus** [comment]
[LogOutput.log](https://github.com/user-attachments/files/17693102/LogOutput.log)

Ok here is the log when we reproduced an issue - bscly at this point if you spawn on the opposite side of big red equals almost all PMC spawn and die around big red until you get there. 

Although it seems that it spawns everyone as far as possible...

```
[Info   :DanW-QuestingBots] Trying to spawn 7 PMC group(s)...
[Info   :DanW-QuestingBots] Found furthest spawn point (-323.9, 1.2, -225.6) that is 842.8434m from other players
[Info   :DanW-QuestingBots] Spawning PMC group at (-323.9, 1.2, -225.6), (-323.9, 1.1, -222.1)...
[Info   :DanW-QuestingBots] Found furthest spawn point (-313.9, 0.9, -88.7) that is 817.1686m from other players
[Info   :DanW-QuestingBots] Spawning PMC group at (-313.9, 0.9, -88.7), (-310.0, 1.0, -94.5), (-306.1, 1.0, -89.4)...
[Info   :DanW-QuestingBots] Found furthest spawn point (-223.2, 1.1, -147.5) that is 731.1353m from other players
[Info   :DanW-QuestingBots] Spawning PMC group at (-223.2, 1.1, -147.5), (-218.7, 1.1, -148.4)...
[Info   :DanW-QuestingBots] Found furthest spawn point (-221.7, -0.7, -3.4) that is 726.7925m from other players
[Info   :DanW-QuestingBots] Spawning PMC group at (-221.7, -0.7, -3.4)...
[Info   :DanW-QuestingBots] Found furthest spawn point (-197.2, 1.5, -233.5) that is 720.9213m from other players
[Info   :DanW-QuestingBots] Spawning PMC group at (-197.2, 1.5, -233.5), (-196.0, 1.5, -234.6)...
[Info   :DanW-QuestingBots] Found furthest spawn point (-154.4, 1.0, -128.1) that is 660.5378m from other players
[Info   :DanW-QuestingBots] Spawning PMC group at (-154.4, 1.0, -128.1), (-151.8, 1.3, -129.8), (-161.6, 3.2, -126.5)...
[Info   :DanW-QuestingBots] Found furthest spawn point (-134.2, -1.4, 38.1) that is 644.6697m from other players
[Info   :DanW-QuestingBots] Spawning PMC group at (-134.2, -1.4, 38.1), (-132.9, -1.7, 39.9)...
```

**Comment by dwesterwick** [comment]
This looks like an incompatibility with Fika. I'll have to look into it.

**Comment by dwesterwick** [comment]
Could you please see if the [0.8.1 (Alpha 1) release](https://github.com/dwesterwick/SPTQuestingBots/releases/tag/0.8.1-alpha1) fixes the problem?

**Comment by Microbicus** [comment]
Yeah. It seems like so. It also greatly changed overall raid experience xD
I never played it without Fika (my group moved from BSG Tarkov directly to SPT with Fika) so I didn't really know how it is supposed to be in bare SPT.
Thank you!

**Comment by Microbicus** [comment]
Yeah. It seems like so. It also greatly changed overall raid experience xD
I never played it without Fika (my group moved from BSG Tarkov directly to SPT with Fika) so I didn't really know how it is supposed to be in bare SPT.
Thank you!

**Comment by dwesterwick** [comment]
FYI - I included this fix in the full [0.8.1 release](https://github.com/dwesterwick/SPTQuestingBots/releases/tag/0.8.1).
